pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/RoaringBitSet.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.common.util.collections;
+
+import java.util.BitSet;
+import java.util.stream.IntStream;
+import org.roaringbitmap.BitSetUtil;
+import org.roaringbitmap.RoaringBitmap;
+
+/**
+ *  A RoaringBitmap-backed BitSet implementation.
+ */
+@SuppressWarnings("all")
+public class RoaringBitSet implements Cloneable {
+    private static final long serialVersionUID = 1L;
+
+    private RoaringBitmap roaringBitmap;
```

Comment:
```suggestion
    private final RoaringBitmap roaringBitmap;
```

Reply:
it can not be final, see `clone`

Reply:
clone should return a new instance

Reply:
```java
    public Object clone() {
        try {
            RoaringBitSet result = (RoaringBitSet) super.clone();
            result.roaringBitmap = roaringBitmap.clone();
            return result;
        } catch (CloneNotSupportedException e) {
            throw new InternalError(e);
        }
    }
```

The ci checks requires that we `must` call `super.clone`.
So I cannot just 
```java
return new RoaringBitSet(roaringBitmap.clone());
```

Reply:
> The ci checks requires that we must call super.clone.

It's better to fix that bad rule used in CI. There are ways to skip the rule for specific methods.